app.py

```diff
@@ -1,64 +1,142 @@
-from flask_api import FlaskAPI 
+from flask_api import FlaskAPI
 from flask import request
+from flask import jsonify
+from flask import json
+from werkzeug.exceptions import HTTPException, BadRequest
 
 
 app = FlaskAPI(__name__)
 
 RESULT = {
-    'language' : 'Python',
-    'framework' : 'Flask',
-    'website' : 'Facebook',
-    'editor' : 'vscode'        
+    'language': 'Python',
+    'framework': 'Flask',
+    'website': 'Facebook',
+    'editor': 'vscode'
 }
 
+
+@app.errorhandler(HTTPException)
+def handle_exception(e):
+    """Return JSON instead of HTML for HTTP errors."""
+    # start with the correct headers and status code from the error
+    response = e.get_response()
+    # replace the body with JSON
+    response.data = json.dumps({
+        "code": e.code,
+        "name": e.name,
+        "description": e.description,
+    })
+    response.content_type = "application/json"
+    return response
+
+
+@app.errorhandler(BadRequest)
+def handle_bad_request(e):
+    return jsonify({
+        "code": e.code,
+        "name": e.name,
+        "description": e.description}), 400
+
+
 @app.route('/', methods=['GET'])
 def get_value():
     key = request.args.get('key')
-    
+
     if not key:
-        return "You forgot key"
-    
+        raise BadRequest("You forgot key!")
+
     if key not in RESULT:
-        return "Key does not exist"
-           
-    return "value: %s " % (RESULT[key])   
-  
+        raise BadRequest("Key does not exist")
+
+    return "value: %s " % (RESULT[key])
+
+
+@app.route('/data', methods=['PUT'])
+def add_default_keys():
+    keys_added = []
+
+    if not request.args:
+        raise BadRequest("You forgot key!")
+
+    for param, key in request.args.items():
+        if not key:
+            raise BadRequest("You forgot key!")
+
+        if key in RESULT:
+            raise BadRequest("Key does exist")
+
```

Comment:
This block certainly violates `ACID` principle

Reply:
em chưa hiểu ACID ạ.
Nhưng hôm nọ em đọc thấy PUT để thêm phần tử hoặc thay thế phần tử cũ, nhưng mà task yêu cầu là thêm phần tử nên em nghĩ là nếu đã có key ở đó rồi thì ko add thêm key trùng tên vào nữa ạ?